### ISSUE TITLE
dune 1 reborn

### ISSUE THREAD

**Issue by gwlegion** [elyosh/scummvm-cryo#2]
we just started a new project for dune 1 remake .

can you contact me https://discord.gg/4qa6XXKkjB

**Comment by stsp** [comment]
That's a really great news, if you ask me.
But your URL asks to register.
Will your project be available somewhere
on a more open resource?

**Comment by gwlegion** [comment]
for now, i've just started to make a team, discord i only a way to talk about it between dev. after that i think we'll open a github

i've started this projetmxt only a week ago... there is much work before starting coding

please join us

Le 17 mars 2021 13:27:59 GMT+01:00, Stas Sergeev ***@***.***> a écrit :
>That's a really great news, if you ask me.
>But your URL asks to register.
>Will your project be available somewhere
>on a more open resource?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/elyosh/scummvm-cryo/issues/2#issuecomment-801041894

-- 
Envoyé de mon téléphone Android avec K-9 Mail. Excusez la brièveté.

**Comment by stsp** [comment]
I am not sure I have time to join
this (even though I contributed a
hsq decompressor to some previous
reborn project), but since I like this
game, some small donation is a due
if you have some progress. :)
So don't hesitate to set up a donations
page.

**Comment by gwlegion** [comment]
At the moment, we don't plan to ask for donation ... we've just started the project... and i don't think we will ask in the future ...but if the time come we will talk about it with all members of the team

but please concider joining us ...
even if you five the project only a few time, you can help us understand your work and make us go faster.

**Comment by stsp** [comment]
It may entirely be possible that
once you put something to look
into, people might got interested
and try to implement some missing
bit.
I think the main problem with
previous attempts was that people
got to know how to render the game's
gfx, but no one knew how to extract
the game scripts. If you have something
in that direction, that would definitely
be a very good motivator. :)

PS: donations never hurt, they motivate.

**Comment by stsp** [comment]
At the very least, you need to write up
a development plan. Please document
what you can borrow from the prev attempts,
and what you plan to work on your own.
Please write all the tasks you have,
small and big, and maybe people will
choose the task that fits their motivation
and schedule.

**Comment by gwlegion** [comment]
to be clear, i'm not a dev ... i'm a fan ...
my work is to find people motivated by the project.
At the moment, dev are talking about the good method to achieve what we want ...
for now, we plant to reimplement the current engine with current gfaphics, and after, adding new functionality

i think we could save some time using all prev attempts, to understand how the game works ... and if i understand correctly, your attempt to make an engine is quite close to our first step

**Comment by gwlegion** [comment]
please concider helping us to make a better task planing, come talk with dev

**Comment by stsp** [comment]
> and if i understand correctly, your attempt to make an engine is quite close to our first step

There is nothing "my" here - I am
just passing by. :) All I did for this
project, was to write the first C version
of a hsq unpacker on some forum
where people were discussing the
hsq format.

**Comment by stsp** [comment]
There is an owner of this project here somewhere,
and I replied simply because I was subscribed to
this project.

**Comment by gwlegion** [comment]
ho sorry, i don't noticed you're not  elyosh  XD

**Comment by KOLANICH** [comment]
1. It is a good idea to publish everything you have already done as early as possible.
2. You may find Kaitai Struct useful for your task.

**Comment by sonicpp** [comment]
Hi guys,
if you want make project for scummvm, then it would be great to contact original authors of this project. This is just fork of (discontinued) project hosted at SF here: [https://sourceforge.net/projects/dunerevival/](https://sourceforge.net/projects/dunerevival/) (the code itself is here: [https://svn.code.sf.net/p/dunerevival/code/](https://svn.code.sf.net/p/dunerevival/code/)).

I managed to convert this project from svn to git (to keep the project history), updated it for scummvm v1.9.0 and published it here: [https://github.com/sonicpp/dunerevival-code](https://github.com/sonicpp/dunerevival-code) - I prefer to keep the history and credit to original authors.

I also rewrote the hsq utility (dont know the history of the hsq, but I think some guy Roman (not sure) wrote the assebler version, then @stsp rewrote it to C?) and added support for compressing. I played with that for some time and can say that my compression util gives same binary results as what did Cryo with their program - the util is here: [https://github.com/sonicpp/Dune-game-translations/blob/master/utils/hsq.c](https://github.com/sonicpp/Dune-game-translations/blob/master/utils/hsq.c).

If you want to continue the crummvm project, I might be interested in as well.

edit: I also started with documentation some time ago: [https://github.com/sonicpp/Dune-game-translations/tree/master/files](https://github.com/sonicpp/Dune-game-translations/tree/master/files)

**Comment by gwlegion** [comment]
hi sonicpp ... i'm already working with a member of dune revival... and we saw your code .
please concider joining us on discord.
ATM, we plan how to make what we want, and after, we'll acquire every tool needed (github, trello ... no matter what is needed)

**Comment by stsp** [comment]
> dont know the history of the hsq, but I think some guy Roman (not sure) wrote the assebler version,
> then @stsp rewrote it to C?

At that time there was already some
rough C++ port of Roman Dolejsi's asm
code. I just wanted to make the code more
portable and properly document all caveats.
And yet I've failed with that task initially:
a few years later I discovered the ability
of pattern repeating on decompression,
and submitted this patch:
https://sourceforge.net/p/dunerevival/patches/1/
The most "interesting" part of it is:
```
+	    /* duplicate the previously extracted data block.
+	     * The data block extracted most recently, can be
+	     * duplicated more than once, precisely ceil(count/offset)
+	     * times. */
+            dup_pos = dst + didx - offset;
+            while (count) {
+	        word len = MIN(count, offset);
+	        memcpy(dst + didx, dup_pos, len);
+	        didx += len;
+	        count -= len;
+	    }
 	}
```

I looked into your code now and I don't
see if you handle such repetitions. Anyway,
this was so long ago... :)
